sniffles/petabi_rule_writer/petabi_rule_writer.py

```diff
@@ -185,16 +196,66 @@ def check_pcre_compile(re):
 # Formats background traffic rule to a petabi rule format.
 # There can only be 1 background traffic rule in a whole rule file.
 # Output: foramtted background traffic rule
-def formatBackgroundTrafficRule(background_traffic):
+def formatBackgroundTrafficRule(background_traffic, protocol_dist=None):
     bgTrafficInfo = []
     bgTrafficInfo.append("    <traffic_stream")
     bgTrafficInfo.append("typets=\"BackgroundTraffic\"")
     bgTrafficInfo.append("percentage=\"" + background_traffic + "\"")
+    if protocol_dist:
+        protocol_dict = protocolPercentage(protocol_dist)
+        for protocol in protocol_dict:
+            distribution = protocol + "=\"" + str(protocol_dict[protocol]) + \
+                           "\""
+            bgTrafficInfo.append(distribution)
     bgTrafficInfo.append(">\n")
     background_rule = ' '.join(bgTrafficInfo)
     return background_rule
 
 
+# Label protocol percentage with given distribution.
+# Wildcard (* symbol) is used to denote even percentage among wildcards.
+# Wildcard protocol will not be shown in the rule, but will be applied
+# in sniffles. When all protocols are given a percentage and their sum
+# does not add up to 100, they are treated as ratio.
+# Output: Dictionary containing protocol percentage
+def protocolPercentage(protocol_dist):
+    protocol_percent = OrderedDict()
+    total = 0
+    num_wildcard = 0
```

Comment:
I'm not seeing where wildcards are handled, specifically.  I think the extra code below (where my other comment is) is accomplishing that.  But probably need to make that more clear.  Also, if this variable is not needed, I think we can remove it.